### PR TITLE
Make YAML configuration work for Bootstrap Config

### LINKED ISSUE #9973: Spring-Cloud-Config-Client Configuration not working
The settings for your Spring-Cloud-Config-Client  are ignored when you configure them in your application.yml (using quarkus-config-yaml extension). Configuring them in application.properties are working normally.

Quarkus Version: 1.5.1.Final

**Expected behavior**
Spring-Cloud-Config-Client configuration should also work in application.yml (like in quarkus 1.4)

**Actual behavior**
Spring-Cloud-Config-Client configuration in application.yml is ignored

quarkus:
..application:
....name: xxx
..spring-cloud-config:
....enabled: true
....url: http://localhost:8889
....fail-fast: true

does not work

quarkus.application.name=xxx
quarkus.spring-cloud-config.enabled=true
quarkus.spring-cloud-config.url=http://localhost:8889
quarkus.spring-cloud-config.fail-fast=true

works

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #9973

The idea here to allow an extension to register which `ConfigSourceProvider` can work in Bootstrap Config, because we (correctly) don't do auto-discovery when building the Bootstrap Config

### CODE REVIEW

**Comment by gsmet** [comment]
@geoand it looks safe enough for a backport, WDYT?

**Comment by geoand** [comment]
@gsmet yeah, it's completely safe

File: core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalBootstrapConfigProviderBuildItem.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * Build item to use when an additional {@link ConfigSourceProvider} needs to be
+ * registered to the Bootstrap Config setup.
+ * This is needed because during Bootstrap Config setup, we don't auto-discover providers
+ * but we do want for example the YAML provider to be enabled.
+ */
+public final class AdditionalBootstrapConfigProviderBuildItem extends MultiBuildItem {
```
Comment (gastaldi) [inline core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalBootstrapConfigProviderBuildItem.java:13]: Maybe it's me, but the name is kinda confusing. Maybe something like `AdditionalConfigSourceProviderBuildItem` ?

Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalBootstrapConfigProviderBuildItem.java:13]: You have a point, I'll make it `AdditionalBootstrapConfigSourceProviderBuildItem`

File: core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalBootstrapConfigSourceProviderBuildItem.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * This is needed because during Bootstrap Config setup, we don't auto-discover providers
+ * but we do want for example the YAML provider to be enabled.
+ */
+public final class AdditionalBootstrapConfigSourceProviderBuildItem extends MultiBuildItem {
+
+    private final String providerClassName;
+
+    public AdditionalBootstrapConfigSourceProviderBuildItem(String providerClassName) {
+        this.providerClassName = providerClassName;
+    }
```
Comment (gastaldi) [inline core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalBootstrapConfigSourceProviderBuildItem.java:19]:
```suggestion
    public AdditionalBootstrapConfigSourceProviderBuildItem(String providerClassName) {
        this.providerClassName = providerClassName;
    }

    public AdditionalBootstrapConfigSourceProviderBuildItem(Class<? extends ConfigSourceProvider> providerClass) {
        this.providerClassName = providerClass.getName();
    }

```

**Review by gastaldi — APPROVED** [review]
LGTM. I added a minor suggestion as an attempt to make it more typesafe, but feel free to ignore :)

File: core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalBootstrapConfigSourceProviderBuildItem.java (hunk 2 above)
Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalBootstrapConfigSourceProviderBuildItem.java:19]: We generally don't pass classes around in BuildItems

**Review by dmlloyd — APPROVED** [review]

**Comment by famod** [comment]
@geoand This doesn't cover custom `ConfigSourceProviders`, right?

We use a custom one to merge multiple yaml files (from maven submodules) that also contain `quarkus` properties and had to realize that any build-time properties from those yaml files are ignored.

**Comment by geoand** [comment]
@famod no it does not. But this PR was only for bootstrap config anyway, not for build-time config.

**Comment by famod** [comment]
Ah, now I understand. Thanks!

**Comment by geoand** [comment]
:+1:

**Pull request merged**